node/config.go

```diff
@@ -152,6 +153,10 @@ type Config struct {
 
 	// Logger is a custom logger to use with the p2p.Server.
 	Logger log.Logger `toml:",omitempty"`
+
+	staticNodesWarning     bool
+	trustedNodesWarning    bool
+	oldGethResourceWarning bool
```

Comment:
Why not use `sync.Once` for these? Would be cleaner imho and would get rid of the global lock.

Reply:
I tried this, but it's not a good idea because some code shallow-copies Config structs. That doesn't work with sync.Once, it contains sync.Mutex.